### PR TITLE
Modernize SiStripMonitorClient

### PR DESCRIPTION
**Pull request by tvami**
#### PR description:

Since I touched the package and gave a lot of warnings about the Legacy modules, 
 * I moved everything to a `::one::` module
 * merged the header files to the cc files
 * moves the es consume into the initializer list

#### PR validation:

Code compiles w/o warning.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by tvami** [comment]
type bugfix,trk

**Comment by tvami** [comment]
@cmsbuild , please test

File: DQM/SiStripMonitorClient/plugins/SiStripAnalyser.cc [hunk 1]
```diff
@@ -1,37 +1,92 @@
...
 #include <sstream>
 #include <string>
 
+class SiStripWebInterface;
+class SiStripFedCabling;
+class SiStripDetCabling;
+class SiStripClassToMonitorCondData;
+class FEDRawDataCollection;
+
+class SiStripAnalyser : public edm::one::EDAnalyzer<> {
```
Comment (mmusich) [inline DQM/SiStripMonitorClient/plugins/SiStripAnalyser.cc:43]: Is this the right fix?
I think this should be moved to `DQMEDHarvester` instead.

File: DQM/SiStripMonitorClient/plugins/SiStripCertificationInfo.cc [hunk 2]
```diff
@@ -1,30 +1,79 @@
...
+#include <string>
+#include <vector>
+#include <map>
 
-#include "DataFormats/Histograms/interface/DQMToken.h"
+class SiStripDetCabling;
 
-//Run Info
-#include "CondFormats/RunInfo/interface/RunInfo.h"
+class SiStripCertificationInfo : public edm::one::EDAnalyzer<> {
```
Comment (mmusich) [inline DQM/SiStripMonitorClient/plugins/SiStripCertificationInfo.cc:28]: Is this the right fix?
I think this should be moved to `DQMEDHarvester` instead.

File: DQM/SiStripMonitorClient/plugins/SiStripDaqInfo.cc [hunk 3]
```diff
@@ -22,6 +25,52 @@
...
 #include <vector>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <map>
+
+class SiStripFedCabling;
+class TrackerTopology;
+
```
Comment (mmusich) [inline DQM/SiStripMonitorClient/plugins/SiStripDaqInfo.cc:36]: Is this the right fix?
I think this should be moved to `DQMEDHarvester` instead.

File: DQM/SiStripMonitorClient/plugins/SiStripDcsInfo.cc [hunk 4]
```diff
@@ -1,20 +1,90 @@
...
+#include <string>
+#include <vector>
+#include <map>
+#include <unordered_map>
+
+class SiStripDetVOff;
+class SiStripDetCabling;
+class RunInfo;
+
+class SiStripDcsInfo : public edm::one::EDAnalyzer<> {
```
Comment (mmusich) [inline DQM/SiStripMonitorClient/plugins/SiStripDcsInfo.cc:36]: Is this the right fix?
I think this should be moved to `DQMEDHarvester` instead.

**Comment by tvami** [comment]
OK, I can make them `DQMEDHarvester`, thanks Marco for the feedback!

**Comment by mmusich** [comment]
assign trk-dpg

> DQMHistoSizes: Histogram memory added: -39.132000000000005 KiB( 50 files compared)
DQMHistoSizes: changed ( 1000.0,... ): -1.131 KiB SiStrip/EventInfo
DQMHistoSizes: changed ( 140.53 ): -0.352 KiB SiStrip/MechanicalView
DQMHistoSizes: changed ( 4.22 ): -4.852 KiB SiStrip/MechanicalView

By the way the comparison in the DQM bin-by-bin does not look good (see [link](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-09-27-2300+b5d35e/52995/dqm-histo-comparison-summary.html))

-1 fwiw.

**Comment by tvami** [comment]
I decoupled the two changes in this PR, as the migration to `DQMEDHarvester` is non-trivial. I get the following error when I simply move to it

```

>> Compiling edm plugin /data/vami/projects/DQM/CMSSW_12_6_X_2022-09-27-1100/src/DQM/SiStripMonitorClient/plugins/SiStripCertificationInfo.cc
/data/vami/projects/DQM/CMSSW_12_6_X_2022-09-27-1100/src/DQM/SiStripMonitorClient/plugins/SiStripCertificationInfo.cc:37:8: error: virtual function 'virtual void SiStripCertificationInfo::endLuminosityBlock(const edm::LuminosityBlock&, const edm::EventSetup&)' overriding final function
   37 |   void endLuminosityBlock(edm::LuminosityBlock const& lumiSeg, edm::EventSetup const& iSetup);
      |        ^~~~~~~~~~~~~~~~~~
In file included from /data/vami/projects/DQM/CMSSW_12_6_X_2022-09-27-1100/src/DQM/SiStripMonitorClient/plugins/SiStripCertificationInfo.cc:9:
/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/DQMServices/Core/interface/DQMEDHarvester.h:132:8: note: overridden function is 'virtual void DQMEDHarvester::endLuminosityBlock(const edm::LuminosityBlock&, const edm::EventSetup&)'
  132 |   void endLuminosityBlock(edm::LuminosityBlock const &, edm::EventSetup const &) final{};
      |        ^~~~~~~~~~~~~~~~~~
In file included from /cvmfs/cms-ib.cern.ch/nweek-02752/slc7_amd64_gcc10/external/gcc/10.3.0-84898dea653199466402e67d73657f10/include/c++/10.3.0/memory:83,
                 from /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/FWCore/Framework/interface/FunctorESHandleExceptionFactory.h:24,
                 from /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/FWCore/Framework/interface/EventSetupRecord.h:42,
                 from /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/FWCore/Framework/interface/EventSetupRecordImplementation.h:26,
                 from /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/CalibTracker/Records/interface/SiStripDependentRecords.h:4,
                 from /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/CalibTracker/Records/interface/SiStripDetCablingRcd.h:1,
                 from /data/vami/projects/DQM/CMSSW_12_6_X_2022-09-27-1100/src/DQM/SiStripMonitorClient/plugins/SiStripCertificationInfo.cc:2:
/cvmfs/cms-ib.cern.ch/nweek-02752/slc7_amd64_gcc10/external/gcc/10.3.0-84898dea653199466402e67d73657f10/include/c++/10.3.0/bits/unique_ptr.h: In instantiation of 'typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...) [with _Tp = SiStripCertificationInfo; _Args = {const edm::ParameterSet&}; typename std::_MakeUniq<_Tp>::__single_object = std::unique_ptr<SiStripCertificationInfo, std::default_delete<SiStripCertificationInfo> >]':
/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/FWCore/Framework/interface/maker/MakeModuleHelper.h:39:40:   required from 'static std::unique_ptr<_Tp> edm::MakeModuleHelper<Base>::makeModule(const edm::ParameterSet&) [with T = SiStripCertificationInfo; Base = edm::one::EDProducerBase]'
/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/FWCore/Framework/interface/maker/WorkerMaker.h:82:107:   required from 'std::shared_ptr<edm::maker::ModuleHolder> edm::WorkerMaker<T>::makeModule(const edm::ParameterSet&) const [with T = SiStripCertificationInfo]'
/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/FWCore/Framework/interface/maker/WorkerMaker.h:76:40:   required from here
/cvmfs/cms-ib.cern.ch/nweek-02752/slc7_amd64_gcc10/external/gcc/10.3.0-84898dea653199466402e67d73657f10/include/c++/10.3.0/bits/unique_ptr.h:962:30: error: invalid new-expression of abstract class type 'SiStripCertificationInfo'
  962 |     { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/vami/projects/DQM/CMSSW_12_6_X_2022-09-27-1100/src/DQM/SiStripMonitorClient/plugins/SiStripCertificationInfo.cc:28:7: note:   because the following virtual functions are pure within 'SiStripCertificationInfo':
   28 | class SiStripCertificationInfo : public DQMEDHarvester {
      |       ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /data/vami/projects/DQM/CMSSW_12_6_X_2022-09-27-1100/src/DQM/SiStripMonitorClient/plugins/SiStripCertificationInfo.cc:9:
/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-09-27-1100/src/DQMServices/Core/interface/DQMEDHarvester.h:166:16: note:     'virtual void DQMEDHarvester::dqmEndJob(dqm::legacy::DQMStore::IBooker&, dqm::legacy::DQMStore::IGetter&)'
  166 |   virtual void dqmEndJob(DQMStore::IBooker &, DQMStore::IGetter &) = 0;
      |                ^~~~~~~~~
```

**Review by mmusich — CHANGES_REQUESTED** [review]

**Comment by mmusich** [comment]
this commit https://github.com/cms-sw/cmssw/commit/40ce9a88c7083ffcabc10edd842538d897bcdf93 should address my https://github.com/cms-sw/cmssw/pull/39518#pullrequestreview-1127916371 comments

**Comment by tvami** [comment]
Thanks Marco! I included your commit. 

I also removed the change in the scripts, however I see this
https://github.com/cms-sw/cmssw/pull/39518/files#diff-6c8707826386170053faf1ed45f98db106967576c345280443c9b830cef288c9

so maybe we can let https://github.com/cms-sw/cmssw/pull/39573 in and then I'll remove on the top of that?

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by mmusich** [comment]
seems tests got stuck here?

**Comment by tvami** [comment]
> seems tests got stuck here?

@smuzaffar ?

**Comment by perrotta** [comment]
@cmsbuild , please test

**Comment by mmusich** [comment]
Is this PR really making the relvals take so long, or is there a problem with the bot?

**Comment by tvami** [comment]
I see this in the Jenkins
```
15:37:08 [parameterized-trigger] Properties file update-das-queries did not exist.
15:37:08 Not triggering due to missing file - did you archive it as a build artifact ?
15:37:08 Triggering projects: [update-das-queries](https://cmssdt.cern.ch/jenkins/job/update-das-queries/)
15:37:08 New run name is 'PR cms-sw/cmssw#39518 el8_amd64_gcc10'
15:37:08 [WS-CLEANUP] Deleting project workspace...
15:37:08 [WS-CLEANUP] Deferred wipeout is used...
15:37:08 [WS-CLEANUP] done
15:37:08 Finished: SUCCESS
```
so it seems it's not actually running

**Comment by tvami** [comment]
@cmsbuild , please abort

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by aandvalenzuela** [comment]
please test

- Tests need to be restarted due to an issue in cvmfs. Sorry for the inconvenience!

**Comment by tvami** [comment]
@mmusich now that my other PR got merged, should I rebase it? (there is no conflict)

**Comment by mmusich** [comment]
> DQMHistoTests: Total failures: 3

Now tests are ~ cleaner

> now that my other PR got merged, should I rebase it? (there is no conflict)

I think it would be cleaner, I honestly don't know how git will resolve the conflict internally.

**Comment by emanueleusai** [comment]
+1

**Comment by tvami** [comment]
> I think it would be cleaner, I honestly don't know how git will resolve the conflict internally.

I rebased, but I still see the `100755 → 100644` change... not sure if we care or not?

**Comment by mmusich** [comment]
> I rebased, but I still see the 100755 → 100644 change... 

did you change the file permissions (i.e. now it is executable)?

**Comment by tvami** [comment]
> did you change the file permissions (i.e. now it is executable)?

Oh I guess yes, I needed that for the unit test I think

**Comment by mmusich** [comment]
> Oh I guess yes, I needed that for the unit test I think

so it's expected. I guess this is ready for merge

**Comment by mmusich** [comment]
@sroychow @connorpa would you mind to sign?

**Comment by sroychow** [comment]
+1

File: DQM/SiStripMonitorClient/plugins/SiStripDaqInfo.cc [hunk 5]
```diff
@@ -22,6 +25,53 @@
 #include <sstream>
 #include <string>
 #include <vector>
+#include <iostream>
```
Comment (perrotta) [inline DQM/SiStripMonitorClient/plugins/SiStripDaqInfo.cc:28]: Duplicates:
```suggestion
```

File: DQM/SiStripMonitorClient/plugins/SiStripDaqInfo.cc [hunk 6]
```diff
@@ -1,18 +1,21 @@
...
 #include "CondFormats/RunInfo/interface/RunInfo.h"
-
-// FED cabling and numbering
+#include "CondFormats/SiStripObjects/interface/SiStripFedCabling.h"
 #include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+#include "DQM/SiStripMonitorClient/interface/SiStripUtility.h"
+#include "DQM/SiStripCommon/interface/SiStripFolderOrganizer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
```
Comment (perrotta) [inline DQM/SiStripMonitorClient/plugins/SiStripDaqInfo.cc:10]: Duplicate:
```suggestion
```

File: DQM/SiStripMonitorClient/plugins/SiStripDcsInfo.cc [hunk 7]
```diff
@@ -1,20 +1,90 @@
...
+#include "DQM/SiStripMonitorClient/interface/SiStripUtility.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "CalibFormats/SiStripObjects/interface/SiStripDetCabling.h"
+#include "CalibTracker/Records/interface/SiStripDetCablingRcd.h"
+#include "CondFormats/RunInfo/interface/RunInfo.h"
+#include "CondFormats/DataRecord/interface/RunSummaryRcd.h"
+#include "CondFormats/DataRecord/interface/SiStripCondDataRecords.h"
+#include "CondFormats/SiStripObjects/interface/SiStripDetVOff.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
```
Comment (perrotta) [inline DQM/SiStripMonitorClient/plugins/SiStripDcsInfo.cc:11]: Duplicate:
```suggestion
```

**Comment by tvami** [comment]
@perrotta I fixed the duplicate includes, thanks for noticing that!

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@sroychow @emanueleusai  can you please re-sign?

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
@sroychow last updates were just cleanings: I think you can sign once again fror @cms-sw/trk-dpg-l2

**Comment by sroychow** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**